src/SFA.DAS.ApprenticeAccounts/Infrastructure/EventDispatcher.cs

```diff
@@ -0,0 +1,26 @@
+﻿using MediatR;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.ApprenticeAccounts.Infrastructure
+{
+    public interface IEventDispatcher
+    {
+        public Task Dispatch(INotification message);
+    }
+
+    public class NullEventDispatcher : IEventDispatcher
+    {
+        public Task Dispatch(INotification message) => Task.CompletedTask;
+    }
+
+    public class EventDispatcher : IEventDispatcher
+    {
+        private readonly IMediator _mediator;
+
```

Comment:
This will probably suffer from the same problems as ApprenticeCommitments. The Message will be published outside of  the transaction

Reply:
We need to understand exactly what is causing the transaction problem in CMAD API.  Once we know what's going wrong and have fixed it there, we can apply the same fix here.

We're not publshing any events from accounts yet anyway.

Reply:
That issue is now fixed in CMAD, so we should be able to resolve it here.

Reply:
Updated to use the UnitOfWork publisher